app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -1354,15 +1399,72 @@ private void populatePlacesForAnotherLocation(
                 }));
     }
 
+    @SuppressLint("CheckResult")
+    public void loadPlacesDataAsync(List<Place> placeList, LatLng curLatLng) {
+        List<Place> places = new ArrayList<>(placeList);
+        int batchSize = 150;
+        List<Place> processedPlaces = new ArrayList<>();
```

Comment:
With the current implementation, too many requests will be sent to the Wikimedia server, making it send HTTP error 429.
So, the loop should wait for each request to get its results before doing the next request.
But refreshing the map can be asynchronous.